arangod/Transaction/Helpers.cpp

```diff
@@ -402,9 +404,17 @@ std::string transaction::helpers::makeIdFromCustom(CollectionNameResolver const*
   TRI_ASSERT(id.isCustom() && id.head() == 0xf3);
   TRI_ASSERT(key.isString());
 
-  uint64_t cid = encoding::readNumber<uint64_t>(id.begin() + 1, sizeof(uint64_t));
+  auto cid = encoding::readNumber<uint64_t>(id.begin() + 1, sizeof(uint64_t));
 
   std::string resolved = resolver->getCollectionNameCluster(cid);
+  if (resolved == "_unknown") {
```

Comment:
This is so bad

Reply:
Are you really gonna compare against that magic string?  This is a code smell

Reply:
No, you're right of course. This was originally just an experiment whether it helped at all, and I've failed to clean it up properly. And I'd still classify this PR as experimental.

However, the whole thing is just a hack, and we should never need to load the plan during a velocypack dump; but that is a larger fix and will need to be done separately.